packages/aic/src/routes.ts

```diff
@@ -0,0 +1,164 @@
+      return await key.sign(msg)
+    },
+    verifyDidSig: crypto.verifyDidSig,
+  }
+}
+// const literalSchema = z.union([z.string(), z.number(), z.boolean(), z.null()])
+// type Literal = z.infer<typeof literalSchema>
+// type Json = Literal | { [key: string]: Json } | Json[]
+// const jsonSchema: z.ZodType<Json> = z.lazy(() =>
+//   z.union([literalSchema, z.array(jsonSchema), z.record(jsonSchema)]),
+// )
+
+// const Patch = z.tuple([
+//   z.literal('put').or(z.literal('del')),
+//   z.string().array(),
+//   jsonSchema,
+// ])
+// type Patch = z.infer<typeof Patch>
+
+// const Diff = z.object({
+//   prev: z.string().regex(/[2-7a-z]{4}-[2-7a-z]{3}-[2-7a-z]{4}-[2-7a-z]{2}/),
+//   patches: Patch.array(),
+//   // base58btc alphabet 123456789 ABCDEFGH JKLMN PQRSTUVWXYZ abcdefghijk mnopqrstuvwxyz
+//   key: z.string().regex(/did:key:z[1-9A-HJ-NP-Za-km-z]+/),
+//   sig: z.string().regex(/z[1-9A-HJ-NP-Za-km-z]+/),
+// })
+// type Diff = z.infer<typeof Diff>
+
+const router = express.Router()
+router.use('/favicon.ico', express.static('static/favicon.ico'))
+
+router.all('/tid', async (req, res) => {
+  res.send(
+    await sign(
+      {
+        tid: tid(),
+        key: (await consortiumCrypto()).did(),
+        sig: '',
+      },
+      await consortiumCrypto(),
+    ),
+  )
+})
+
+router.get(
+  '/:pid([234567abcdefghijklmnopqrstuvwxyz]{16})/',
+  async function (req, res) {
+    // AIC git
+    //
+    // Retreve the latest tick from the databace
+    // pass the tick and curent tid to the aic lib for signing
+    const did = 'did:aic:' + req.params.pid
+    const row = await res.locals.db.tickForDid(did) // retreve latest tick
+    const prev_tick = row ? JSON.parse(row.tick) : null
+    const signedDoc = await updateTick(
+      did,
+      tid(),
+      null, // candidateDiff: this is a get no updates
+      prev_tick,
+      await consortiumCrypto(),
+    )
+    res.type('json').send(signedDoc)
+  },
+)
+
+router.post(
+  '/:pid([234567abcdefghijklmnopqrstuvwxyz]{16})/',
+  async function (req, res) {
+    // extract from post
+    const did = 'did:aic:' + req.params.pid
+    const candidateDiff = req.body
+
+    // extract from DB
+    const row = await res.locals.db.tickForDid(did)
+    const prevTid = row ? row.tid : null
+    const prev_tick = row ? JSON.parse(row.tick) : null
+
+    // aic lib will do the doc update
+    const newTick = await updateTick(
+      did, // did from the URL
+      tid(), // curent time (tid)
+      candidateDiff, // diff that was posted/put
+      prev_tick, // tick from db for did:aic
+      await consortiumCrypto(), // server's aic key
+    )
+
+    if ('error' in newTick) {
+      // error return from updateTick
+      // type ErrorMessage = { error: string; cause?: ErrorMessage; [index: string]: Value }
+      // if there is an error property it is an error don't store in DB
+      // also send the last tick that is still valid
+      res.status(200)
```

Comment:
should this be a 200?

& this is sorta my point on error handling.

`if ('error' in newTick) {` _works_, but it's not the most idiomatic way of doing it. & if it's not captured by either the type system or throw/catches, I can see errors slipping by & causing issues